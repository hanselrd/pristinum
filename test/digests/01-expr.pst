func main() @void:
    nil;
    true;
    false;
    'c';
    1;
    3.14159;
    "string";
    x;
    main();
    main(1);
    main(1, 2);
    (u64) 1;
    p x .``;
    p x ->``;
    `*p;
    `&p;
    size? u64;
    align? u64;
    x ++`;
    x --`;
    x -`;
    x ~`;
    x !`;
    x y +``;
    x y -``;
    x y *``;
    x y /``;
    x y %``;
    x y **``;
    x y <<``;
    x y >>``;
    x y &``;
    x y |``;
    x y ^``;
    x y ==``;
    x y !=``;
    x y <``;
    x y >``;
    x y <=``;
    x y >=``;
    x y &&``;
    x y ||``;
end