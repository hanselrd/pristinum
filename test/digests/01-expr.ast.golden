PProgram
    [ PUFunction
        ( FFunction
            ( IDENT "main" ) [] TVoid
            [ SExpr ENil
            , SExpr ETrue
            , SExpr EFalse
            , SExpr
                ( EChar 'c' )
            , SExpr
                ( EInt 1 )
            , SExpr
                ( EDouble 3.14159 )
            , SExpr
                ( EString "string" )
            , SExpr
                ( EIdent
                    ( IDENT "x" )
                )
            , SExpr
                ( ECall
                    ( IDENT "main" ) []
                )
            , SExpr
                ( ECall
                    ( IDENT "main" )
                    [ EInt 1 ]
                )
            , SExpr
                ( ECall
                    ( IDENT "main" )
                    [ EInt 1
                    , EInt 2
                    ]
                )
            , SExpr
                ( ECast
                    ( TIdent
                        ( IDENT "u64" )
                    )
                    ( EInt 1 )
                )
            , SExpr
                ( EAccess
                    ( EIdent
                        ( IDENT "p" )
                    )
                    ( EIdent
                        ( IDENT "x" )
                    )
                )
            , SExpr
                ( EAccess
                    ( EIdent
                        ( IDENT "p" )
                    )
                    ( EIdent
                        ( IDENT "x" )
                    )
                )
            , SExpr
                ( EDeref
                    ( EIdent
                        ( IDENT "p" )
                    )
                )
            , SExpr
                ( ERef
                    ( EIdent
                        ( IDENT "p" )
                    )
                )
            , SExpr
                ( ESizeof
                    ( TIdent
                        ( IDENT "u64" )
                    )
                )
            , SExpr
                ( EAlignof
                    ( TIdent
                        ( IDENT "u64" )
                    )
                )
            , SExpr
                ( EIncr
                    ( EIdent
                        ( IDENT "x" )
                    )
                )
            , SExpr
                ( EDecr
                    ( EIdent
                        ( IDENT "x" )
                    )
                )
            , SExpr
                ( ENeg
                    ( EIdent
                        ( IDENT "x" )
                    )
                )
            , SExpr
                ( EBNot
                    ( EIdent
                        ( IDENT "x" )
                    )
                )
            , SExpr
                ( ELNot
                    ( EIdent
                        ( IDENT "x" )
                    )
                )
            , SExpr
                ( EAdd
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( ESubtract
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( EMultiply
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( EDivide
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( EMod
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( EPower
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( EBShl
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( EBShr
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( EBAnd
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( EBOr
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( EBXor
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( EEqual
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( ENotEqual
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( ELess
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( EGreater
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( ELessEqual
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( EGreaterEqual
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( ELAnd
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            , SExpr
                ( ELOr
                    ( EIdent
                        ( IDENT "x" )
                    )
                    ( EIdent
                        ( IDENT "y" )
                    )
                )
            ]
        )
    ]